src/proj.rs

```diff
@@ -0,0 +1,791 @@
+//! Projective transformations of images.
+
+use image::{Pixel, GenericImageView, ImageBuffer};
+use crate::definitions::{Clamp, Image};
+
+use crate::math::cast;
+use conv::ValueInto;
+use std::ops::Mul;
+use rayon::prelude::*;
+
+#[derive(Copy, Clone, Debug)]
+enum TransformationClass {
+    Translation,
+    Affine,
+    Projection
+}
+
+/// A 2d Projective transformation, stored as a row major 3x3 matrix.
+/// Transformations combine by multiplication.
+///
+/// I.e. to define rotation around image center
+/// ```
+/// use imageproc::proj::*;
+/// let (cx,cy) = (320.0,240.0);
+/// let c_rotation = Proj::trans(cx,cy)*Proj::rot(30.0)*Proj::trans(-cx, -cy);
```

Comment:
Nit: always insert spaces after commas in tuples, and around operators.

In this case `(320.0, 240.0)` and 'Proj::trans(cx, cy) * Proj::rot(30.0) * Proj::trans(-cx, -cy);`